### PR TITLE
GitHub ssh access

### PR DESCRIPTION
**Pull request by tank157**
Assigning to @cap10morgan for a good review of this code.

If we accept this, we still need to manually clean up the current fleet deploy, remove the old buildkite@.service and buildkite@1 and buildkite-coreos@1 and buildkite-coreos@.service and then deploy this version. But then we should have 3 agents running on coreos.

[Bug Card](https://www.pivotaltracker.com/story/show/92148198)

### CODE REVIEW

File: buildkite@.service [hunk 1]
```diff
@@ -21,6 +21,7 @@ ExecStart=/bin/bash -c 'docker run --name ${CONTAINER} \
   --env BUILDKITE_AGENT_META_DATA=coreos=true \
   --env DOCKERCFG=\"$(curl -s http://${COREOS_PRIVATE_IPV4}:8500/v1/kv/buildkite/dockercfg?raw)\" \
   --env FLEETCTL_ENDPOINT=http://${COREOS_PRIVATE_IPV4}:4001 \
+  --env GITHUB_PRIVATE_KEY=\"$(curl -s http://${COREOS_PRIVATE_IPV4}:8500/v1/kv/github/private-key?raw)\" \
```
Comment (cap10morgan) [inline buildkite@.service]: Can we call this `deploy-private-key` or something else a little more descriptive so we can trace it back to its public counterpart on GitHub more easily?

File: script/deploy [hunk 2]
```diff
@@ -24,17 +24,10 @@ IMAGE_TAG=$(echo $DOCKER_IMAGE | awk -F: '{print $2}')
...
+fleetctl submit buildkite@.service
 
-echo '--- (re-)starting fleet service instances'
-for i in {1..3}; do
-  $fleetctl destroy buildkite@$i
-  $fleetctl start buildkite@$i
-done
+echo '--- (re-)starting fleet service instance'
+fleetctl destroy buildkite@1
+fleetctl start buildkite@1
```
Comment (cap10morgan) [inline script/deploy]: I think we will want 3 of these running once this is merged and used to deploy the agents. Can you put back the for loop?

File: start [hunk 3]
```diff
@@ -2,6 +2,13 @@
 
 set -e
 
+if [[ -n "${GITHUB_PRIVATE_KEY}" ]]; then
+  mkdir /root/.ssh
+  chmod 700 /root/.ssh
```
Comment (cap10morgan) [inline start]: Let's move lines 6 and 7 into the Dockerfile. Then we can assume in here that the `/root/.ssh` dir exists and has the right permissions.

File: script/build [hunk 4]
```diff
@@ -6,8 +6,13 @@ if [[ -z $DOCKER_REPO && -n $1 ]]; then
   DOCKER_REPO=$1
 fi
 
+if [[ -z $DOCKER_REPO ]]; then
+  echo 'ERROR: DOCKER_REPO or command line arg required.'
+  exit 1
+fi
+
```
Comment (cap10morgan) [inline script/build]: We should try getting this from the buildkite@.service file before failing, [like this](https://github.com/democracyworks/consul-coreos/blob/master/script/build#L9-L16).

Comment (ericnormand) [inline script/build]: DONE

**Comment by ericnormand** [comment]
Just as a note for how to run/test that this can deploy on the cluster:

```
> script/build
> docker push ... (just copy it from the message near the end of the output of script/build)
> script/deploy quay.io/democracyworks/buildkite-agent-coreos:<TAG NAME>
```

Then verify it's running:

```
> fleetctl list-units
```

**Comment by tank157** [comment]
A second way to test that I use sometimes when I just want to spin up a single instance is like what Eric suggests, but slightly different:

```
> script/build
> docker push ... (just copy it from the message near the end of the output of script/build)
# edit fleetctl to use the same image tag as you just pushed
> fleetctl submit buildkite-agent-consul@.service
> fleetctl start buildkite-agent-consul@1
```

It's more or less the same thing the deploy script is doing, but just does it to a single CoreOS node. In fact, maybe we can make the deploy script take the number of nodes as a parameter.

**Pull request merged**